ui/admin/app/components/form/target/details/index.js

```diff
@@ -37,4 +42,37 @@ export default class FormTargetComponent extends Component {
   get showDeprecationMessage() {
     return !this.args.model.isNew && this.args.model.worker_filter;
   }
+
+  /**
+   * determines when the update worker filter button should be shown
+   * @type {boolean}
+   */
+  get showUpdateWorkerFilterButton() {
+    return (
+      !this.args.model.isNew &&
+      this.args.model.worker_filter &&
+      !this.updateDeprecatedWorkerFilter
+    );
+  }
+  /**
+   * returns egress_worker_filter value from the model,
+   * but in case of updating the deprecated worker filter, value from worker_filter field is returned
+   * @type {boolean}
+   */
+  get egressWorkerFilterValue() {
+    return this.args.model.egress_worker_filter?.length
+      ? this.args.model.egress_worker_filter
+      : this.args.model.worker_filter;
+  }
+  // =actions
+  @action
+  toggleEgressWorkerFilter() {
+    this.egressWorkerFilterEnabled = !this.egressWorkerFilterEnabled;
+  }
+  // =actions
+  @action
+  updateDeprecatedFilter() {
+    this.updateDeprecatedWorkerFilter = true;
+    this.egressWorkerFilterEnabled = !this.egressWorkerFilterEnabled;
```

Comment:
The toggle's value is derived, not set.  It is enabled whenever `this.args.model.egress_worker_filter` has a value.